File: 49-group-anagrams/group-anagrams.java
class Solution {
  public List<List<String>> groupAnagrams(String[] strs) {
    List<List<String>> output = new ArrayList<>();
    HashMap<String, List<String>> anagrams = new HashMap<String, List<String>>();
    for (int i = 0; i < strs.length; i++) {
      char[] charArray = strs[i].toCharArray();
      Arrays.sort(charArray);
      String sortedStr = new String(charArray);
      if (anagrams.containsKey(sortedStr)) {
        anagrams.get(sortedStr).add(strs[i]);
      } else {
        List<String> temp = new ArrayList<>();
        temp.add(strs[i]);
        anagrams.put(sortedStr, temp);
      }
    }
    for (String asd : anagrams.keySet()) {
      output.add(anagrams.get(asd));
    }
    return output;
  }
}

File: 49-group-anagrams/group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function (strs) {
  const anagrams = new Map();
  for (let i = 0; i < strs.length; i++) {
    const sortedWord = strs[i].split("").sort().join("");

    if (!anagrams.has(sortedWord)) {
      anagrams.set(sortedWord, [strs[i]]);
    } else {
      anagrams.get(sortedWord).push(strs[i]);
    }
  }
  return Array.from(anagrams.values());
};